pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -243,7 +243,7 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
       Set<String> unavailableSegments = entry.getValue();
       numUnavailableSegments += unavailableSegments.size();
       brokerResponse.addException(QueryException.getException(QueryException.SERVER_SEGMENT_MISSING_ERROR,
-          String.format("Find unavailable segments: %s for table: %s", unavailableSegments, tableName)));
+          String.format("Found unavailable segments for table {%s}: {%s}", tableName, unavailableSegments)));
```

Comment:
```suggestion
          String.format("Found unavailable segments for table %s: %s", tableName, unavailableSegments)));
```

Reply:
Consider adding number of unavailable segments, and only log up to 10 segments. If there are more than 10 segments unavailable, append `...` after the list

Reply:
Done @Jackie-Jiang. Also, given that error path will be very infrequent, I felt it was ok to use stream way to build the string as opposed to StringBuilder and for loop.